src/lib/merkle_ledger_tests/visualizable_ledger.ml

```diff
@@ -1,46 +1,143 @@
 open Core
+open Async
 
-module type Ledger_intf = sig
-  type t
+(** Visualizable_ledger shows a subgraph of a merkle_ledger using Graphviz *)
+module type S = sig
+  type addr
 
-  type hash [@@deriving sexp, eq]
+  type tree
 
-  val max_depth : int
+  type t
 
-  val get_inner_hash_at_addr_exn : t -> Direction.t list -> hash
+  (* Visualize will enumerate through all edges of a subtree with a root
+     initial_address. It will then interpret all of the edges and nodes into an
+     intermediate form that will be easy to write into a dot file *)
+  val visualize : tree -> initial_address:addr -> t
+
+  (* Write will transform the intermediate form generate by visualze and save
+     the results into a dot file *)
+  val write : path:string -> name:string -> t -> unit Deferred.t
 end
 
-module type S = sig
-  type t
+module type Inputs_intf = sig
+  module Key : Merkle_ledger.Intf.Key
+
+  module Balance : Merkle_ledger.Intf.Balance
+
+  module Account : sig
+    include Merkle_ledger.Intf.Account with type key := Key.t
 
-  type hash [@@deriving sexp, eq]
+    val balance : t -> Balance.t
+  end
 
-  type tree =
-    | Leaf of (Direction.t list * hash)
-    | Node of (Direction.t list * hash * tree * tree)
-  [@@deriving sexp, eq]
+  module Hash : Merkle_ledger.Intf.Hash with type account := Account.t
 
-  val to_tree : t -> tree
+  module Location : Merkle_ledger.Location_intf.S
+
+  module Ledger :
+    Merkle_ledger.Base_ledger_intf.S
+    with module Addr = Location.Addr
+     and module Location = Location
+    with type key := Key.t
+     and type key_set := Key.Set.t
+     and type hash := Hash.t
+     and type root_hash := Hash.t
+     and type account := Account.t
 end
 
-module Make (L : Ledger_intf) = struct
-  type t = L.t
+module Make (Inputs : Inputs_intf) :
+  S with type addr := Inputs.Location.Addr.t and type tree := Inputs.Ledger.t =
+struct
+  open Inputs
+
+  module Account = struct
+    include Account
+    include Comparator.Make (Account)
+  end
 
-  type hash = L.hash [@@deriving sexp, eq]
+  type edge = Inner of (Hash.t * Hash.t) | Leaf of (Hash.t * Account.t option)
 
-  type tree =
-    | Leaf of (Direction.t list * hash)
-    | Node of (Direction.t list * hash * tree * tree)
-  [@@deriving sexp, eq]
+  type t = (string * string) list * (string * int) list
 
-  let to_tree t =
-    let rec go i dirs =
-      if i = L.max_depth then Leaf (dirs, L.get_inner_hash_at_addr_exn t dirs)
-      else
-        let hash = L.get_inner_hash_at_addr_exn t dirs in
-        let left = go (i + 1) (dirs @ [Direction.Left]) in
-        let right = go (i + 1) (dirs @ [Direction.Right]) in
-        Node (dirs, hash, left, right)
+  let shorten_string string = String.prefix string 8
```

Comment:
comment to indicate purpose here, like

  "prefix of string for display purposes"

maybe call this "display_prefix_of_string"